pkg/backup/compressor.go

```diff
@@ -0,0 +1,61 @@
+package backup
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/klauspost/compress/zstd"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+)
+
+const extension = "zst"
+
+// Compress compresses a file using zstd.
```

Comment:
Using `zstd` as the compression algorithm and doing the process in stream mode to avoid reading the whole file into memory